File: C programming/Day-6/binary to decimal.c
#include <stdio.h>

int main() {
    int binary, decimal = 0, base = 1, remainder, temp;

    printf("Enter a binary number: ");
    scanf("%d", &binary);

    temp = binary;

    while (temp > 0) {
        remainder = temp % 10;
        decimal = decimal + remainder * base;
        base = base * 2;
        temp = temp / 10;
    }

    printf("Decimal value of %d is %d\n", binary, decimal);
    return 0;
}

File: Java programming/Day-6/binary to decimal.java
import java.util.Scanner;

public class BinaryToDecimal {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long binary = scanner.nextLong();

        long decimal = 0;
        long base = 1;
        long temp = binary;

        while (temp > 0) {
            long lastDigit = temp % 10;
            temp /= 10;
            decimal += lastDigit * base;
            base *= 2;
        }

        System.out.println(decimal);
        scanner.close();
    }
}

